turbopy/core.py

```diff
@@ -25,7 +25,86 @@ class Simulation:
+    Parameters
+    ----------
+    input_data : `dict`
+        This dictionary contains all parameters needed to set up a
+        turboPy simulation. Each key describes a section, and the value
+        is another dictionary with the needed parameters for that
+        section.
+
+        Expected keys are:
+
+        ``"Grid"``
+            Dictionary containg parameters needed to defined the grid.
+            Currently only 1D grids are defined in turboPy.
+
+            The expected parameters are:
+
+            - ``"N"`` | {``"dr"`` | ``"dx"``} :
+                The number of grid points (`int`) | the grid spacing
+                (`float`)
+            - ``"min"`` | ``"x_min"`` | ``"r_min"`` :
+                The coordinate value of the minimum grid point (`float`)
+            - ``"max"`` | ``"x_max"`` | ``"r_max"`` :
+                The coordinate value of the maximum grid point (`float`)
+
+        ``"Clock"``
+            Dictionary of parameters needed to define the simulation
+            clock.
+
+            The expected parameters are:
+
+            - ``"start_time"`` :
+                The time for the start of the simulation (`float`)
+            - ``"end_time"`` :
+                The time for the end of the simulation (`float`)
+            - ``"num_steps"`` | ``"dt"`` :
+                The number of time steps (`int`) | the size of the time
+                step (`float`)
+            - ``"print_time"`` :
+                `bool`, optional, default is ``False``
+
+        ``"PhysicsModules"`` : `dict` [`str`, `dict`]
+            Dictionary of PhysicsModules needed for the simulation.
+
+            Each key in the dictionary should map to a PhysicsModule
+            subclass key in the PhysicsModule registry.
+
+            The value is a dictionary of parameters which is passed to
+            the constructor for the PhysicsModule.
+
+        ``"Diagnostics"`` : `dict` [`str`, `dict`], optional
+            Dictionary of Diagnostics needed for the simulaiton.
+
+            Each key in the dictionary should map to a Diagnostic
+            subclass key in the Diagnostic registry.
+
+            The value is a dictionary of parameters which is passed to
+            the constructor for the Diagnostic.
+
+            If the key is not found in the registry, then the key/value
+            pair is interpreted as a default parameter value, and is
+            added to dictionary of parameters for all of the Diagnostic
+            constructors.
+
+        ``"Tools"`` : `dict` [`str`, `dict`], optional
+            Dictionary of ComputeTools needed for the simulation.
+
+            Each key in the dictionary should map to a ComputeTool
+            subclass key in the ComputeTool registry.
+
+            The value is a dictionary of parameters which is passed to
+            the constructor for the ComputeTool.
+
+    Attributes
+    ----------
+    physics_modules : list of PhysicsModule subclass objects
```

Comment:
You can use :class:\`PhysicsModule\` to refer to objects in your code. So, this line should be:

physics_modules: list of :class:\`PhysicsModule\`

the lines below this could be done similarly

Reply:
Thanks! I'm still trying to learn the syntax... this is why reviews are so helpful!

Reply:
Wasn't sure if you would want me to just make the changes myself or comment and let you make the changes. Let me know for the future.

Reply:
Is there a similar notation that should be used to refer to specific member functions?

Reply:
Yes, just :class:\`ClassName.member_function\`.

Reply:
Thanks! I was trying :method:`member_function` and variations of that... 😄

Reply:
In this case I prefer getting the comments, so that I can learn better practices for writing docstrings!